diff --git a/examples/apps/cli/main.c b/examples/apps/cli/main.c
index 2ede597cd..bb6264f83 100644
--- a/examples/apps/cli/main.c
+++ b/examples/apps/cli/main.c
@@ -47,6 +47,12 @@
 
 #include "lib/platform/reset_util.h"
 
+#include <openthread/dataset.h>
+#include <openthread/ip6.h>
+#include <openthread/thread.h>
+#include <stdbool.h>
+#include <openthread/config.h>
+
 /**
  * Initializes the CLI app.
  *
@@ -137,6 +143,7 @@ pseudo_reset:
     instance = otInstanceInitSingle();
 #endif
     assert(instance);
+    bool autoStarted = false;
 
     otAppCliInit(instance);
 
@@ -150,6 +157,15 @@ pseudo_reset:
 
     while (!otSysPseudoResetWasRequested())
     {
+	#if OPENTHREAD_CONFIG_IP6_ENABLE && OPENTHREAD_CONFIG_THREAD_ENABLE
+        if (!autoStarted && otDatasetIsCommissioned(instance))
+        {
+            otIp6SetEnabled(instance, true);
+            otThreadSetEnabled(instance, true);
+            autoStarted = true;
+        }
+    #endif
+
         otTaskletsProcess(instance);
         otSysProcessDrivers(instance);
     }
